pkg/agent/execbridge.go

```diff
@@ -52,20 +67,26 @@ func (iface *execPluginInterface) Request(
 		iface.runner.recordResourceChange(*lastPermit, resp.Permit, iface.runner.global.metrics.schedulerApprovedChange)
 	}
 
-	successful := func() bool {
+	successful, responseType := func() (bool, scalingResponseType) {
```

Comment:
I think you don't need to keep "successful" separate, you can just check responseType Below in the callback.

Reply:
done